pyanaconda/ui/tui/spokes/network.py

```diff
@@ -176,66 +321,96 @@ def refresh(self, args=None):
         self.redraw()
         self.apply()
 
-    def _configure_network_interface(self, data):
-        devname = data
-        ndata = network.ksdata_from_ifcfg(devname)
-        if not ndata:
-            # There is no ifcfg file for the device.
-            # Make sure there is just one connection for the device.
-            try:
-                nm.nm_device_setting_value(devname, "connection", "uuid")
-            except nm.SettingsNotFoundError:
-                log.debug("can't find any connection for %s", devname)
-                return
-            except nm.MultipleSettingsFoundError:
-                log.debug("multiple non-ifcfg connections found for %s", devname)
-                return
-
-            log.debug("dumping ifcfg file for in-memory connection %s", devname)
-            nm.nm_update_settings_of_device(devname, [['connection', 'id', devname, None]])
-            ndata = network.ksdata_from_ifcfg(devname)
-
-        new_spoke = ConfigureNetworkSpoke(self.data, self.storage, self.payload, ndata)
+    def _ensure_connection_and_configure(self, iface):
+        for device_configuration in self.editable_configurations:
+            if device_configuration.device_name == iface:
+                connection_uuid = device_configuration.connection_uuid
+                if connection_uuid:
+                    self._configure_connection(iface, connection_uuid)
+                    return
+                else:
+                    device_type = nm_client.get_device_by_iface(iface).get_device_type()
+                    connection = get_default_connection(iface, device_type)
+                    connection_uuid = connection.get_uuid()
+                    log.debug("adding default connection %s for %s", connection_uuid, iface)
+                    persistent = False
+                    data = (iface, connection_uuid)
+                    nm_client.add_connection_async(connection, persistent, None,
+                                                   self._default_connection_added_cb, data)
+                    return
+        log.error("device configuration for %s not found", iface)
+
+    def _default_connection_added_cb(self, client, result, data):
+        client.add_connection_finish(result)
+        iface, connection_uuid = data
+        log.debug("added default connection %s for %s", connection_uuid, iface)
+        self._configure_connection(iface, connection_uuid)
+
+    # TODO store the value in network module instead of ifcfg file?
+    # SetConnectionOnboot(uuid, value)
+    # GetConnectionOnboot(uuid)
+    def _get_onboot(self, connection_uuid):
+        return get_onboot_from_ifcfg(connection_uuid)
+
+    def _set_onboot(self, connection_uuid, onboot):
+        return set_onboot_in_ifcfg(connection_uuid, onboot)
+
+    def _configure_connection(self, iface, connection_uuid):
+        connection = nm_client.get_connection_by_uuid(connection_uuid)
+        ui_data = WiredTUIConfigurationData()
```

Comment:
It looks like the `ui_data` could be created in the `ConfigureNetworkSpoke` and applied in its apply method. You can get the necessary information for the `NetworkSpoke` directly from the spoke instance. Also, I don't like the name `ui_data`. The name suggests that the data has something to do with widgets or something more UI-related.

Reply:
How about renaming ui_data to ui_input_data? Or ui_config_data?

Reply:
https://github.com/rhinstaller/anaconda/pull/1802/commits/09985ac46747baa3300a8ad27cdec231024acebc

Reply:
Looks great. Thanks!